chain/get_ancestors.go

```diff
@@ -124,3 +127,43 @@ func CollectAtMostNTipSets(ctx context.Context, iterator *TipsetIterator, n uint
 	}
 	return ret, nil
 }
+
+// FindCommonAncestor returns the common ancestor of the two tipsets pointed to
+// by the input iterators.  If they share no common ancestor errIterComplete
+// will be returned.
+func FindCommonAncestor(oldIter, newIter *TipsetIterator) (types.TipSet, error) {
```

Comment:
nit: "old" and "new" are not relevant concepts within this function; names like "left" and "right" would avoid implying semantics that don't exist.